### PR TITLE
Updates and grammar fixes to README.md

### PR DESCRIPTION
**Pull request by joelhans**
##### Summary

I realized recently that our main README is out of date in regards to the database engine, which prompted these changes. Aside from changing references from `round-robin` to `database engine`, 

I also took this opportunity to integrate grammatical fixes in certain areas. Some of those fixes were pulled directly from #7073, which I'm going to close now. Others are just from me working my way down from the top of the page and making fixes and improving consistency.

I'll work on other fixes in future PRs.

##### Component Name

docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,7 +126,8 @@ Netdata has a quite different approach to monitoring.
 Netdata is a monitoring agent you install on all your systems. It is:
 
 -   a **metrics collector** - for system and application metrics (including web servers, databases, containers, etc)
--   a **time-series database** - all stored in memory (does not touch the disks while it runs)
+-   a **database engine** - recent metrics stored in memory while historical metrics "spilled" to disk for
```
Comment (amoss) [inline README.md]: either "spilled" to "spill" for correct tense, or "while" -> "with" to remove the tense.

File: README.md [hunk 2]
```diff
@@ -172,7 +174,7 @@ Release v1.18.0 contains 5 new collectors, 16 bug fixes, 27 improvements, and 20
 
 The **database engine** is now the default method of storing metrics in Netdata. You immediately get more efficient and configurable long-term metrics storage without any work on your part. By saving recent metrics in RAM and "spilling" historical metrics to disk for long-term storage, the database engine is laying the foundation for many more improvements to distributed metrics.
 
-We even have a [tutorial](https://docs.netdata.cloud/docs/tutorials/longer-metrics-storage/) on switching to the database engine and getting the most from it. Or, just read up on [how performant](https://docs.netdata.cloud/database/engine/#evaluation) the database engine really is.
+We even have a [tutorial](https://docs.netdata.cloud/docs/tutorials/longer-metrics-storage/) on switching to the database engine and getting the most from it. Or, just read up on [how performant](https://docs.netdata.cloud/database/engine/#evaluation) the database engine is.
```
Comment (amoss) [inline README.md]: Is ending in a preposition deliberate? - it seems ok in the informal case.

File: README.md [hunk 3]
```diff
@@ -376,16 +378,16 @@ This is what you should expect from Netdata:
...
--   **Scales to infinity** - you can install it on all your servers, containers, VMs and IoTs. Metrics are not centralized by default, so there is no limit.
--   **Several operating modes** - Autonomous host monitoring (the default), headless data collector, forwarding proxy, store and forward proxy, central multi-host monitoring, in all possible configurations. Each node may have different metrics retention policy and run with or without health monitoring.
+-   **1s granularity** - The highest possible resolution for all metrics.
+-   **Unlimited metrics** - Netdata collects all the available metrics—the more, the better.
+-   **1% CPU utilization of a single core** - It's unbelievably optimized.
+-   **A few MB of RAM** - The low-memory round-robin option uses 25MB RAM, and you can [resize it](database/).
+-   **Minimal disk I/O** - While running, Netdata only writes historical metrics and reads `error` and `access` logs.
+-   **Zero configuration** - Netdataa auto-detects everything, and can collect up to 10,000 metrics per server out of the box.
```
Comment (amoss) [inline README.md]: Double a on the "Netdataa"

File: README.md (hunk 2 above)
Comment (joelhans) [reply README.md]: I'm a pretty informal writer! But I did fix this to make it read a little nicer.

File: README.md (hunk 1 above)
Comment (joelhans) [reply README.md]: Went "while" -> "with"

**Pull request merged**